### DISCUSSION TITLE
Feature Request: Portable mode

### DISCUSSION THREAD

**Discussion by flvinny** [chrislo27/PolyrhythmMania#9] [Ideas]
# Feature Request: Portable mode
A portable mode which can be enabled via a launch argument, akin to the one in RHRE. Changes the location where the  `.polyrhythmmania/` directory can be found to be in the same directory as the `PolyrhythmMania.jar` executable.

## Justification
Paraphrasing the RHRE docs: "[...] useful for portable flash drives."
Personally, I have PrMania saved on an external hard drive, and would rather to have my preferences stored alongside the game to it being in my home folder, in a different hard drive.

**Comment by chrislo27** [comment]
The temporary files folder is stored in the `tmp` directory inside `.polyrhythmmania` and is used for things including decoding music. Rapid I/O access is important for these files. Therefore, the temp directory may still be placed on the same drive as the user home, going with the reasonable assumption that the user home is on the boot drive most of the time.

Additionally, asking the user if they want to automatically migrate the `.polyrhythmmania` directory over when a new version is released would be handy.

**Comment by chrislo27** [comment]
Elevated to an issue in #11.

**Comment by chrislo27** [comment]
Implemented as of [v1.0.1](https://github.com/chrislo27/PolyrhythmMania/releases/tag/v1.0.1).
